### PR TITLE
update clusterctl versions to latest patches

### PR DESCRIPTION
**Pull request by ykakarap**
Update `clusterctl` version in all the upgrade e2e tests to the newest patch. 
Ref: https://github.com/kubernetes-sigs/cluster-api/issues/6056

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-main.yaml [hunk 1]
```diff
@@ -47,7 +47,7 @@ periodics:
           - name: GINKGO_FOCUS
             value: "\\[clusterctl-Upgrade\\]"
           - name: INIT_WITH_BINARY
-            value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.23/clusterctl-{OS}-{ARCH}
+            value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.26/clusterctl-{OS}-{ARCH}
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-main.yaml:50]:
```suggestion
            value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.25/clusterctl-{OS}-{ARCH}
```

If I got it correctly v0.3.25 is not affected, so there will be no v0.3.26

Comment (sbueringer) [reply config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-main.yaml:50]: Same below

**Comment by sbueringer** [comment]
/lgtm
/assign @fabriziopandini 

@fabriziopandini Up to you if you want to merge now or after the release. It doesn't really matter, given that the jobs are already failing now.

**Comment by ykakarap** [comment]
/hold cancel

The required releases are now available.

**Pull request merged**